rust/src/fvm/machine.rs

```diff
@@ -0,0 +1,276 @@
+use std::convert::{TryFrom, TryInto};
+use std::sync::Mutex;
+
+use cid::Cid;
+use ffi_toolkit::{catch_panic_response, raw_ptr, rust_str_to_c_str, FCPResponseStatus};
+use futures::executor::block_on;
+use fvm::call_manager::DefaultCallManager;
+use fvm::executor::{ApplyKind, DefaultExecutor, Executor};
+use fvm::machine::DefaultMachine;
+use fvm::{Config, DefaultKernel};
+use fvm_ipld_car::load_car;
+use fvm_shared::blockstore::Blockstore;
+use fvm_shared::{clock::ChainEpoch, econ::TokenAmount, message::Message, version::NetworkVersion};
+use lazy_static::lazy_static;
+use log::info;
+
+use super::blockstore::{CgoBlockstore, FakeBlockstore, OverlayBlockstore};
+use super::externs::CgoExterns;
+use super::types::*;
+use crate::util::api::init_log;
+
+pub type CgoExecutor = DefaultExecutor<
+    DefaultKernel<DefaultCallManager<DefaultMachine<OverlayBlockstore<CgoBlockstore>, CgoExterns>>>,
+>;
+
+lazy_static! {
+    static ref ENGINE: fvm::machine::Engine = fvm::machine::Engine::default();
+}
+
+/// Note: the incoming args as u64 and odd conversions to i32/i64
+/// for some types is due to the generated bindings not liking the
+/// 32bit types as incoming args
```

Comment:
This is very suspicious, and easy to get wrong. What were the issues you had here?